src/foam/core/XML.js

```diff
@@ -202,7 +207,11 @@ foam.CLASS({
 
     function outputProperty_(o, p) {
       if ( ! this.propertyPredicate(o, p ) ) return;
-      if ( ! this.outputDefaultValues && p.isDefaultValue(o[p.name]) ) return;
+      // don't output default values unless value is defined and outputDefinedValues set to true
+      if ( ( ! this.outputDefaultValues && p.isDefaultValue(o[p.name]) ) &&
+        ( this.outputDefinedValues && ! o.hasOwnProperty(p.name) ) ) {
+        return;
```

Comment:
I think this is easier to understand (and also more efficient):

if ( this.outputDefinedValues ) {
  if ( ! o.hasOwnProperty(p.name) ) return;
} else if ( this.outputDefaultValues ) {
  if ( p.isDefaultValue(o[p.name]) ) return;
}